src/control/server/mgmt_svc.go

```diff
@@ -470,6 +470,59 @@ func (svc *mgmtSvc) smdReplace(ctx context.Context, req *mgmtpb.SmdQueryReq) (*m
 	}, nil
 }
 
+func (svc *mgmtSvc) smdIdentify(ctx context.Context, req *mgmtpb.SmdQueryReq) (*mgmtpb.SmdQueryResp, error) {
+	req.Rank = uint32(system.NilRank)
+	rank, device, err := svc.smdQueryDevice(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	if device == nil {
+		return nil, errors.Errorf("smdIdentify on %s did not match any devices", req.Uuid)
+	}
+
+	// TODO: Make sure the device is a VMD device (Here??)
+
+	srvs, err := svc.harness.FilterInstancesByRankSet(fmt.Sprintf("%d", rank))
+	if err != nil {
+		return nil, err
+	}
+	if len(srvs) == 0 {
+		return nil, errors.Errorf("failed to retrieve instance for rank %d", rank)
+	}
+
+	svc.log.Debugf("calling storage identify on rank %d for %s", rank, req.Uuid)
+
+	dresp, err := srvs[0].CallDrpc(ctx, drpc.MethodIdentifyStorage, &mgmtpb.DevIdentifyReq{
+		DevUuid: req.Uuid,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	drr := &mgmtpb.DevIdentifyResp{}
+	if err = proto.Unmarshal(dresp.Body, drr); err != nil {
+		return nil, errors.Wrap(err, "unmarshal StorageIdentify response")
+	}
+
+	if drr.Status != 0 {
+		return nil, errors.Wrap(drpc.DaosStatus(drr.Status), "smdIdentify failed")
+	}
+
+	return &mgmtpb.SmdQueryResp{
+		Ranks: []*mgmtpb.SmdQueryResp_RankResp{
+			{
+				Rank: rank.Uint32(),
+				Devices: []*mgmtpb.SmdQueryResp_Device{
+					{
+						Uuid:  drr.DevUuid,
+						State: drr.LedState,
+					},
+				},
+			},
+		},
+	}, nil
```

Comment:
This pattern is repeated over and over. It's time to refactor and create a helper to DRY things up.